corehq/pillows/case_search.py

```diff
@@ -75,7 +74,12 @@ def _get_case_properties(doc_dict):
 class CaseSearchPillowProcessor(ElasticProcessor):
     def process_change(self, pillow_instance, change, do_set_checkpoint):
         assert isinstance(change, Change)
-        domain = change.get_document()['domain']
+        if change.metadata is not None:
```

Comment:
do you mean the branching? longer term maybe populate metadata in the reindexing change provider...